app/models/info_request_event.rb

```diff
@@ -384,49 +378,65 @@ def response?
   end
 
   def same_email_as_previous_send?
-    prev_addr = self.info_request.get_previous_email_sent_to(self)
-    curr_addr = self.params[:email]
+    prev_addr = info_request.get_previous_email_sent_to(self)
+    curr_addr = params[:email]
     if prev_addr.nil? && curr_addr.nil?
       return true
     end
     if prev_addr.nil? || curr_addr.nil?
       return false
     end
-    return MailHandler.address_from_string(prev_addr) == MailHandler.address_from_string(curr_addr)
+    MailHandler.address_from_string(prev_addr) == MailHandler.address_from_string(curr_addr)
   end
 
   def json_for_api(deep, snippet_highlight_proc = nil)
     ret = {
-      :id => self.id,
-      :event_type => self.event_type,
+      :id => id,
+      :event_type => event_type,
       # params_yaml has possibly sensitive data in it, don't include it
-      :created_at => self.created_at,
-      :described_state => self.described_state,
-      :calculated_state => self.calculated_state,
-      :last_described_at => self.last_described_at,
-      :incoming_message_id => self.incoming_message_id,
-      :outgoing_message_id => self.outgoing_message_id,
-      :comment_id => self.comment_id,
+      :created_at => created_at,
+      :described_state => described_state,
+      :calculated_state => calculated_state,
+      :last_described_at => last_described_at,
+      :incoming_message_id => incoming_message_id,
+      :outgoing_message_id => outgoing_message_id,
+      :comment_id => comment_id,
 
       # TODO: would be nice to add links here, but alas the
       # code to make them is in views only. See views/request/details.html.erb
       # perhaps can call with @template somehow
     }
 
-    if self.is_incoming_message? || self.is_outgoing_message?
-      ret[:display_status] = self.display_status
+    if is_incoming_message? || is_outgoing_message?
+      ret[:display_status] = display_status
     end
 
-    if !snippet_highlight_proc.nil?
-      ret[:snippet] = snippet_highlight_proc.call(self.search_text_main(true))
+    if snippet_highlight_proc
+      ret[:snippet] = snippet_highlight_proc.call(search_text_main(true))
     end
 
     if deep
-      ret[:info_request] = self.info_request.json_for_api(false)
-      ret[:public_body] = self.info_request.public_body.json_for_api
-      ret[:user] = self.info_request.user_json_for_api
+      ret[:info_request] = info_request.json_for_api(false)
+      ret[:public_body] = info_request.public_body.json_for_api
+      ret[:user] = info_request.user_json_for_api
+    end
+
+    ret
+  end
+
+  def set_calculated_state!(state)
+    unless calculated_state == state
+      self.calculated_state = state
+      self.last_described_at = Time.now
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.